packages/components/back-top/src/BackTop.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <transition name="ix-fade">
+    <div v-if="visible" class="ix-back-top" @click.stop="handleClick">
+      <slot>
+        <ix-icon name="vertical-align-top" />
+      </slot>
+    </div>
+  </transition>
+</template>
+
+<script lang="ts">
+import type { SetupContext } from 'vue'
+import type { BackTopProps } from './types'
+
+import { defineComponent, ref, onMounted, onBeforeUnmount } from 'vue'
+import { IxIcon } from '@idux/components/icon'
+import { PropTypes, withUndefined, isString, isFunction, on, off, raf } from '@idux/cdk/utils'
+import { easeInOutQuad } from '@idux/components/core/utils'
+import { useGlobalConfig } from '@idux/components/core/config'
+import throttle from 'lodash/throttle'
+
+const getTarget = (target: string | (() => HTMLElement)) => {
+  if (isString(target)) {
+    return document.querySelector<HTMLElement>(target)
+  }
+
+  if (isFunction(target)) {
+    return target()
+  }
+
+  console.warn(`props: target should be a string or a function.`)
+  return null
+}
+
+const getCurrentScrollTop = (currTarget: Window | HTMLElement) => {
+  if (currTarget === window) {
+    return window.pageYOffset || document.body.scrollTop || document.documentElement.scrollTop
+  }
+
+  return (currTarget as HTMLElement).scrollTop
+}
+
+const scrollTo = (y: number, duration: number, container: HTMLElement | Window) => {
+  const scrollTop = getCurrentScrollTop(container)
+  const startTime = Date.now()
+  const change = y - scrollTop
+
+  const frameFunc = () => {
+    const time = Date.now() - startTime
+    const nextScrollTop = easeInOutQuad(time > duration ? duration : time, scrollTop, change, duration)
+
+    if (container === window) {
+      ;(container as Window).scrollTo(window.pageXOffset, nextScrollTop)
+    } else {
+      ;(container as HTMLElement).scrollTop = nextScrollTop
+    }
+
+    if (time < duration) {
+      raf(frameFunc)
+    }
+  }
+
+  raf(frameFunc)
+}
+
+export default defineComponent({
+  name: 'IxBackTop',
+  components: { IxIcon },
+  props: {
+    target: withUndefined(PropTypes.oneOfType([PropTypes.string, PropTypes.func])),
+    duration: PropTypes.number,
+    visibilityHeight: PropTypes.number,
+  },
+  emits: ['click'],
+  setup(props: BackTopProps, { emit }: SetupContext) {
+    const backTopConfig = useGlobalConfig('backTop')
+    const eventType = 'scroll'
+    const visible = ref(false)
+    const container = ref<HTMLElement | Window>(window)
+
+    const handleScroll = () => {
+      visible.value = getCurrentScrollTop(container.value) >= (props.visibilityHeight ?? backTopConfig.visibilityHeight)
+    }
+
+    const handleClick = (event: Event) => {
```

Comment:
```suggestion
    const handleClick = (evt: MouseEvent) => {
```

event 默认是个全局变量。。。有时候会踩坑。